[AzureMonitorExporter] Integration of OpenTelemetry Resource Detectors (#37837)
Azure/azure-sdk-for-net: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md

### Features Added

* Added `Resource` to traces, logs, and metrics with default configuration.
  ([#37837](https://github.com/Azure/azure-sdk-for-net/pull/37837))

> * Added package `OpenTelemetry.ResourceDetectors.Azure` to introduce OpenTelemetry Resource detectors `AppServiceResourceDetector` and `AzureVMResourceDetector`.

Reviewer: ```suggestion
* Added resource detection for `Azure App Service` and `Azure VM` environment. 
```

Reviewer: Do we need a readme update on how to override these?

Author: Yes, we should a follow up PR.

Author: @TimothyMothra Could you please help here once this PR gets merged?

Author: This is covered already in readme - https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore#adding-custom-resource

I tested and it works.

Author: Should we document `builder.Services.AddOpenTelemetry().ConfigureResource(resourceBuilder => resourceBuilder.AddService("service-name"));` which is common for all 3 signals.

  ([#37837](https://github.com/Azure/azure-sdk-for-net/pull/37837))

### Breaking Changes

### Bugs Fixed

### Other Changes

* Update OpenTelemetry dependencies
  ([#37837](https://github.com/Azure/azure-sdk-for-net/pull/37837))
  - OpenTelemetry 1.5.1
  - OpenTelemetry.Extensions.Hosting 1.5.1
  - OpenTelemetry.Instrumentation.AspNetCore 1.5.1-beta.1
  - OpenTelemetry.Instrumentation.Http 1.5.1-beta.1

>   - OpenTelemetry.Instrumentation.SqlClient 1.5.1-beta.1

Reviewer 2: should also include a similar message in the Exporter Changelog

Author: Only for OpenTelemetry package?

Reviewer 2: yup. that's what we've been doing :)
